app/routes/goal_routes.py

```diff
@@ -0,0 +1,128 @@
+from flask import Blueprint, jsonify, make_response, request, abort
+from werkzeug.exceptions import RequestEntityTooLarge
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+import requests
+
+
+goal_bp = Blueprint("goals", __name__,url_prefix="/goals")
+
+# HELPER FUNCTION
+def valid_int(number, parameter_type):
+    try:
+        number = int(number)
+    except:
+        abort(400, {"error":f"{parameter_type} must be an int"})
+
+def get_goal_from_id(goal_id):
+    valid_int(goal_id, "goal_id")
+    return Goal.query.get_or_404(goal_id)
+
+# GOAL ROUTES
+
+# Create a goal
+@goal_bp.route("", methods=["POST"])
+def create_goal():
+    request_body = request.get_json()
+    if "title" not in request_body:
+        return {"details": "Invalid data"}, 400
+
+    new_goal = Goal(
+        title = request_body["title"]
+    )
+
+    db.session.add(new_goal)
+    db.session.commit()
+
+    response_body = {"goal":new_goal.to_dict_goal()}
+
+    return jsonify(response_body), 201
+
+# Read all goals
+@goal_bp.route("", methods=["GET"])
+def read_all_goals():
+    goals = Goal.query.all()
+    goals_response = []
+    for goal in goals:
+        goals_response.append(
+            goal.to_dict_goal()
+        )
+
+    return jsonify(goals_response), 200
+
+# Get one goal
+@goal_bp.route("/<goal_id>", methods=["GET"])
+def read_one_goal(goal_id):
+    goal = get_goal_from_id(goal_id)
+    response_body = {"goal":goal.to_dict_goal()}
+    return jsonify(response_body),200
+
+# Update goal
+@goal_bp.route("/<goal_id>", methods=["PUT"])
+def update_goal(goal_id):
+    goal = get_goal_from_id(goal_id)
+    request_body = request.get_json()
+    if "title" not in request_body:
+        return {"message":"Request requires a title"}, 400
+    else:
+        goal.title = request_body["title"]
+       
+        db.session.commit()
+
+        response_body = {"goal":goal.to_dict_goal()}
+        return jsonify(response_body), 200
```

Comment:
Consider moving this code outside the else to enhance readability
```suggestion
    goal.title = request_body["title"]
       
    db.session.commit()

    response_body = {"goal":goal.to_dict_goal()}
    return jsonify(response_body), 200
```